pkgs/development/libraries/libayatana-appindicator/default.nix

```diff
@@ -26,12 +26,16 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ pkg-config autoreconfHook gtk-doc gobject-introspection python2 python2Packages.pygtk dbus-glib ];
 
   buildInputs =
-    lib.lists.optional (gtkVersion == "2") libayatana-indicator-gtk2
-    ++ lib.lists.optional (gtkVersion == "3") libayatana-indicator-gtk3;
-
-  propagatedBuildInputs =
-    lib.lists.optionals (gtkVersion == "2") [ gtk2 libdbusmenu-gtk2 ]
-    ++ lib.lists.optionals (gtkVersion == "3") [ gtk3 libdbusmenu-gtk3 ];
+    lib.lists.optionals (gtkVersion == "2") [
+      libayatana-indicator-gtk2
+      libdbusmenu-gtk2
+      gtk2
+    ]
+    ++ lib.lists.optionals (gtkVersion == "3") [
+      libayatana-indicator-gtk3
+      libdbusmenu-gtk3
+      gtk3
+    ];
```

Comment:
we have these in Requires so we still need to propagate them `Requires: dbusmenu-glib-0.4 gtk+-3.0`. I'm not sure this is the solution we should take. Changes like this are basically everywhere

Reply:
Hm.. I could at least build the packages I depend on without propagating.
The problem with propagating is that there is no distinction between run time an build time dependencies so the gtk `-dev` output ends up in the run time closure.